include/algorithms/synthesis/dd_synthesis.hpp

```diff
@@ -56,6 +69,10 @@ namespace syrec {
 
         auto unifyPath(dd::mEdge src, dd::mEdge const& current, TruthTable::Cube::Set const& p1SigVec, TruthTable::Cube::Set const& p2SigVec, bool const& changePaths, std::unique_ptr<dd::Package<>>& dd) -> dd::mEdge;
         auto shiftingPaths(dd::mEdge const& src, dd::mEdge const& current, std::unique_ptr<dd::Package<>>& dd) -> dd::mEdge;
+
+        auto decoder(TruthTable::CubeMap const& codewords, std::unique_ptr<dd::Package<>>& dd) -> void;
```

Comment:
I believe the DD package is not required as a parameter here and in all the functions above.

Reply:
So based on the above, I just need to remove the dd parameter from decoder right?

Reply:
Probably.. for now.